### ISSUE TITLE
any plans for including music?

### ISSUE THREAD

**Issue by aleph-bet** [endless-sky/endless-sky#233]
has anyone taken it upon themselves to compose a soundtrack?

**Comment by speeder** [comment]
I offered some time ago to make a music engine for the game, the author said that each person prefers a different kind of music, and he himself don't like music on the game, so... no.

The game don't even support music files, unless you try to shoehorn music into sound effect files.

**Comment by endless-sky** [comment]
It's on the long-term [Development Roadmap](https://github.com/endless-sky/endless-sky/wiki/DevelopmentRoadmap), but it's not something I've put much work into yet.

The only way it would be worth adding music to the game would be if the music were *so* good that people would not just turn it off and play music that's more to their own liking in the background, instead. (At least, that's what I do with almost every game I play.) So that sets a really high bar - the music needs to be able to compete with professional recordings, *and* has to be in a genre that players like.

**Comment by speeder** [comment]
Some professional musicians I know, are willing to make music for small projects.

Usually on two versions: one, that the project is completely free (easy here).

another version is that they can create a full soundtrack worth of music, and then they are allowed to sell it from their site.

Two guys that I know that are into taht stuff are Virt, and Siren

Virt's site: https://virt.bandcamp.com/

Siren has no site that I know of right now, but he did the music for Unreal Tournament, Jazz Jackrabbit, Bejeweled, Deus Ex, and a couple of other games.

**Comment by bobthebike** [comment]
Please make this an absolute no, no.  The vast majority of music in games is horribly repetative sampling, or very well done but limited in the amount of different tunes/songs. once again making it repetative and boring, and or it clashes with the sound effects.   Why not just put on a CD/mp3.
Personaly I Have the sound off in most games anyway.

**Comment by natangold** [comment]
> Personaly I Have the sound off in most games anyway.

So I'm not the only one who does this? Nifty.

**Comment by lkraider** [comment]
Having some good original music brings value by itself.

And you can always mute and play your own if you want.

**Comment by KiLEdEnNis** [comment]
first it'd take hours of music unless you want a repetitive, and even then players prefer their own music, so putting effort in it will hardly pay off

**Comment by lkraider** [comment]
Yes, if you put little effort it will hardly pay off.

**Comment by KiLEdEnNis** [comment]
no I mean if MZ puts a lot of effort in it it will hardly pay off as only like 3 people actually listen to it & 97 just turn it off

**Comment by speeder** [comment]
I will be one of the 3.

Like I said earlier, I could get professionals, that have actual fandoms (as in, people like their game music so much they actually even buy albums and play games they worked in) to help, the question here, from my point of view, specially considering the nature of the project (open source), is not the effort the maintainer needs, it is if people are willing to allow, or not allow, music in the game.

I don't have time now (some bad stuff happened on my personal life), but I could even code the music engine if needed, no effort would be needed from the maintainer.

**Comment by KiLEdEnNis** [comment]
you can try but I still don't see it as a good idea, a side reason is that the game is intended as a small game that any PC can play, adding music makes it a bit heavier
that and the other reasons like hardly anyone appreciates it

**Comment by Alkallid** [comment]
Think some more, maybe some opinions are a bit different from yours.

**Comment by KiLEdEnNis** [comment]
yes I won't stop anyone from making it but I just warn that not many people may like it

**Comment by Alkallid** [comment]
_Think some more._

**Comment by KiLEdEnNis** [comment]
well it may be cool, but there'll have to be several hours of music unless you want 1 repetitive song, we'll see if it gets added sometime

**Comment by endless-sky** [comment]
It really confuses me why people don't just play music of their own choosing while playing games. There are very few games that have music good enough that I'd listen to it over my own mp3 collection.

I do plan to add a "music" engine to the game eventually, but the first thing it will be used for will be to add ambient sounds when you're landed on a planet, to add a bit more atmosphere - e.g. ocean waves on one planet, wind on another, etc.

**Comment by KiLEdEnNis** [comment]
@endless-sky then making .gif planet images (like a sea planet you actually see the moving sea)
maybe one day we'll see that too, and if that'll be added it would be great, but that'd be a whole lot of work

**Comment by warthist** [comment]
First off, I would also like to participate in the music production process for an ES soundtrack, if there was to be any, however I am no professional and I can, by any means, compete with those. Second of all, if people aren't bothered by official game soundtrack playing while they're playing the game (and there are people like that), the raw quality doesn't isn't actually the music's main selling point. If a small indie game, such as ES, attracts players despite its inability to technologically compete with contemporary AAA titles, I don't see why would the game need a most professional soundtrack recording, if it certainly isn't in need of full-HD textures or a complex game engine.

Second, the music styles of individual tracks should be composed primarily with the game's theme in mind, regardless of what styles are now popular and conquering the top 10 international music charts. A game's music score's goal is to emphasize the player's experience (that goes for SFX in general).

Third, as I see there is a divided opinion regarding music ingame, why don't you (meaning the dev) make a survey about this topic and post it in the game's Steam news? That way you get a more specified opinion from a wider range of the player base.

**Comment by TdoubleAV** [comment]
another idea to make the game more unique and deal with the „game music“ problem:

what if the new pilot starts with no music at all.
then you encounter friendly ships and additionally to the „I’m listening to …“ you sometimes get to click a link in the hail to actually listen to the track mentioned (or some annoying pilot plays it over the radio or something like this).

then it get’s added to your library and you can choose to add it to your playlist or not.
so over time you get a collection of hundreds of tracks and the player creates his own personal game soundtrack.

**Comment by Bladewood** [comment]
interesting idea, but it would be pretty confusing for most people.

**Comment by MessyMix** [comment]
@TdoubleAV that sounds awesome af. +1 :D

**Comment by jafdy** [comment]
Not using the system would be much like the game is now. I do like that idea.
What about having the library be persistent across all saves to add replay value (and it would be annoying to start collecting again with every new game).

**Comment by AdamKauffman** [comment]
I think music should only play on the title screen and when you're in a Spaceport. It's more ambient that way. Music in games doesn't always make sense. I like when it's somehow tied to the atmosphere, like a radio in the game that can be turned off. Music also makes sense for the quest "There might be riots".

**Comment by mirror176** [comment]
Music should otherwise be a thing to set the gameplay atmosphere rather than be what is on our top 10 music charts (or whatever the people here commenting they already turn off 'all' game music to listen to instead. Obviously the music should have volume/off type of controls.
  I like the idea of the band music playing; maybe as an album you get and can toggle. Other than some kind of acquirable albums, planets you are near could have radio to tune into making maps unique. Distortion could be considered as a distance from planet and gone when docked.
  I'm also a fan of dynamically chosen music which could be on enemy versus ally zones and whether or not things are contested there. Few games have tried to have music writing change based on the moment but  if I recall fluidsynth was a program created with that as a goal in mind. 
  Ambient planet sounds is definitely an idea I figure this game will head to even if music also comes; its like the docking at a space station kinda feel.

**Comment by TheSilverStone** [comment]
Maybe it would be a good idea to add it only to the menu, and available to change sound in preferences. There could be a setting that changes the music on or off, and the default is off. You could add a little comment that only opens the first time: "Music is in preferences, and the default is off" so that no confusion is caused
